chirp_client.cc

```diff
@@ -0,0 +1,302 @@
+#include "chirp_client.h"
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <grpcpp/grpcpp.h>
+#include <google/protobuf/util/time_util.h>
+
+#include "service.grpc.pb.h"
+
+DEFINE_string(register, "", "register user in key value store");
+DEFINE_string(user, "", "current user of chirp");
+DEFINE_string(chirp, "", "string to chirp out");
+DEFINE_string(reply, "", "string to reply to a given chirp");
+DEFINE_string(read, "", "Reads the chirp thread starting at the given id");
+DEFINE_string(follow, "", "Starts following the given username");
+DEFINE_bool(monitor, false, "Streams new tweets from those currently followed");
+
+void ChirpClient::registeruser(const std::string& user) {
+  grpc::ClientContext context;
+  chirp::RegisterRequest request;
+  chirp::RegisterReply reply;   // Data from server will be updated here
+
+  request.set_username(user);
+  
+  // RPC call
+  grpc::Status status = stub_->registeruser(&context, request, &reply);
```

Comment:
Should be pointing a pointer to request.

Reply:
wait why?